File: ujin/2주차_수학/6603.py
def DFS(L, s):
    if L == 6:
        print(*res)
        return

    else:
        for i in range(s, K):
            res[L] = S[i]
            DFS(L+1,i+1)

while True:
    number = list(map(int, input().split()))
    if number[0] == 0:
        break
    K = number[0]
    S = number[1:]

    res = [0] * 6
    DFS(0, 0)
    print()

'''
K (6<K<13) 개의 수 중 6개의 숫자를 골라 조합하는 문제이다.
문제에서 S의 원소는 오름차순으로 주어진다고 했으므로 따로 sort는 사용하지 않았다.
DFS를 이용해 조합을 구하고자 했고, 이때 중복이 없어야 하므로 DFS를 재귀호출 할 때
s의 값을 i+1을 해줌으로써 이전에 선택한 인덱스의 다음 인덱스부터 for문이 돌도록 했다.
'''

File: syuzzang/2주차수학/6603.js
let fs = require("fs");
let input = fs.readFileSync("./input.txt").toString().trim().split("\n");

input.pop();
input = input.map((v) => v.split(" ").map(Number));

let answer = "";
let firstNum;
let restedNum;

input.forEach((x) => {
  [firstNum, ...restedNum] = x;
  // 로또의 개수와 숫자들을 구조분해할당으로 나눔
  generateLotto(0, []);
  // 함수 초기값설정
});

function generateLotto(L, pick) {
  if (pick.length === 6) {
    answer += `${pick.join(" ")}\n`;
    return;
  }
  // 길이가 6개만 걸러낸다
  for (let i = L; i < firstNum; i++) {
    generateLotto(i + 1, [...pick, restedNum[i]]);
  }
  //  for문을 돌려서 답안 배열에 값을 추가해준다 L이 0이될때까지 반복됨
  // 예를들어 L이 5라고 했을때 5~firstnum까지 반복하고 배열 뒤에 들어감
}

console.log(answer);
